plugins/permission-backend/src/service/PermissionIntegrationClient.ts

```diff
@@ -42,20 +46,30 @@ export type ResourcePolicyDecision = ConditionalPolicyDecision & {
 };
 
 export class PermissionIntegrationClient {
-  private readonly discovery: PluginEndpointDiscovery;
+  private readonly discovery: DiscoveryService;
+  private readonly auth: AuthService;
 
-  constructor(options: { discovery: PluginEndpointDiscovery }) {
+  constructor(options: { discovery: DiscoveryService; auth: AuthService }) {
     this.discovery = options.discovery;
+    this.auth = options.auth;
   }
 
   async applyConditions(
     pluginId: string,
+    credentials: BackstageCredentials,
     decisions: readonly ApplyConditionsRequestEntry[],
-    authHeader?: string,
   ): Promise<ApplyConditionsResponseEntry[]> {
-    const endpoint = `${await this.discovery.getBaseUrl(
-      pluginId,
-    )}/.well-known/backstage/permissions/apply-conditions`;
+    const baseUrl = await this.discovery.getBaseUrl(pluginId);
+    const endpoint = `${baseUrl}/.well-known/backstage/permissions/apply-conditions`;
+
+    const token = this.auth.isPrincipal(credentials, 'none')
```

Comment:
I have a feeling that skipping this checking and always requesting the obo would actually have the intended effect as well. This feels a bit safer and more explicit though